engines/buried/scene_view.cpp

```diff
@@ -0,0 +1,2690 @@
+		initializeTimeZoneAndEnvironment(this, newLocation.timeZone, -1);
+	if (newLocation.environment != oldLocation.environment && newLocation.environment != -2)
+		initializeTimeZoneAndEnvironment(this, newLocation.timeZone, newLocation.environment);
+
+	SceneBase *newScene = constructSceneObject(this, newSceneStaticData, passedLocation);
+
+	if (_currentScene && _currentScene->postExitRoom(this, passedLocation) == SC_DEATH)
+		return false;
+
+	if (!newScene)
+		error("Failed to create scene");
+
+	if (_currentScene) {
+		_currentScene->preDestructor();
+		delete _currentScene;
+		_currentScene = 0;
+	}
+
+	if (newLocation.timeZone != oldLocation.timeZone || newLocation.environment != oldLocation.environment || oldLocation.timeZone < 0)
+		startEnvironmentAmbient(passedLocation.timeZone, passedLocation.environment, newLocation.timeZone, newLocation.environment);
+
+	_currentScene = newScene;
+
+	if (isCyclingEnabled() && newSceneStaticData.cycleStartFrame == -1)
+		flushCycleFrameCache();
+
+	invalidateWindow(false);
+
+	if (_currentScene->preEnterRoom(this, passedLocation) == SC_END_PROCESSING)
+		return true;
+
+	if (_globalFlags.bcCloakingEnabled != 1)
+		((GameUIWindow *)_parent)->_navArrowWindow->updateAllArrows(newScene->_staticData);
+
+	if (newLocation.timeZone != oldLocation.timeZone)
+		((GameUIWindow *)_parent)->changeCurrentDate(newLocation.timeZone);
+
+	invalidateWindow(false);
+
+	_currentScene->postEnterRoom(this, passedLocation);
+
+	_parent->invalidateWindow(false);
+
+	if (((GameUIWindow *)_parent)->_inventoryWindow->isItemInInventory(kItemBioChipAI))
+		playAIComment(newSceneStaticData.location, AI_COMMENT_TYPE_SPONTANEOUS);
+
+	((GameUIWindow *)_parent)->_bioChipRightWindow->sceneChanged();
+
+	return true;
+}
+
+bool SceneViewWindow::jumpToSceneRestore(const Location &newLocation) {
+	Location oldLocation(-2, -2, -2, -2, -2, -2);
+	Location passedLocation(-2, -2, -2, -2, -2, -2);
+
+	if (_infoWindowDisplayed)
+		((GameUIWindow *)getParent())->_inventoryWindow->destroyInfoWindow();
+	if (_bioChipWindowDisplayed)
+		((GameUIWindow *)getParent())->_bioChipRightWindow->destroyBioChipViewWindow();
+	if (_burnedLetterDisplayed)
+		((GameUIWindow *)getParent())->_inventoryWindow->destroyBurnedLetterWindow();
+
+	// Get the static scene data for this new location
+	LocationStaticData newSceneStaticData;
+	if (!getSceneStaticData(newLocation, newSceneStaticData))
+		return false;
+	if (_currentScene)
+		oldLocation = _currentScene->_staticData.location;
+
+	// Clear the live text window
+	if (newLocation.timeZone != oldLocation.timeZone || newLocation.environment != oldLocation.environment)
+		((GameUIWindow *)getParent())->_liveTextWindow->updateLiveText("");
+
+	// If we have a scene, call the pre-transition function
+	if (_currentScene)
+		_currentScene->preExitRoom(this, passedLocation);
+
+	// The original resets the environment upon loading, which is clearly not correct.
+	// We won't.
+
```

Comment:
I'm a bit wary of leaving commented out code permanently in source. If it's demonstrated to be incorrect, then it would be cleaner to simply remove it, and make a note in the in the commit message. And maybe put a 'WORKAROUND:' prefix on the comment indicating that an original problem has been worked around.